senders/mattermost/sender.go

```diff
@@ -0,0 +1,187 @@
+package mattermost
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/moira-alert/moira"
+	"github.com/moira-alert/moira/senders"
+
+	"github.com/mattermost/mattermost-server/v6/model"
+)
+
+// Sender posts messages to Mattermost chat.
+// It implements moira.Sender.
+// You must call Init method before SendEvents method.
+type Sender struct {
+	frontURI string
+	location *time.Location
+	client   Client
+}
+
+// Init configures Sender.
+func (sender *Sender) Init(senderSettings map[string]string, _ moira.Logger, location *time.Location, _ string) error {
```

Comment:
Maybe it would be safer to create function `NewSender` that would create instance of Sender and initialise it. This way user will be able to avoid separation of definition and implementation.

Reply:
Yes, I agree with you. I thought the same, but one thing is stopped me - current design of Sender interface, there is `Init` method in interface. If I add consctructor func `NewSender(settings, logger, location, etc.)` with code of `Init` method right now, then it will be special case for mattermost.Sender (all other Senders now configured in `Init` method). But I also want to fix it, so I suggest to create an issue to refactor all Senders in one Pull Request, how do you think @Tetrergeru @kissken ?

Reply:
one == another?

let's take that ticket to refactor next, good?

@Dimedrolity

Reply:
Yes, I meant in another PR

Reply:
I created [issue](https://github.com/moira-alert/moira/issues/797), now I will do this in another PR